### PR TITLE
[bcl] Move the mono specific code in BitConverter to a partial class. 

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by marek-safar** [comment]
@vargaz could you please rebase it

**Comment by marek-safar** [comment]
@monojenkins build failed

File: mcs/class/corlib/System/BitConverter.cs [hunk 1]
```diff
@@ -2,25 +2,23 @@
...
 namespace System
 {
-    public static partial class BitConverter
+	public static partial class BitConverter
 	{
 		[Intrinsic]
 		public static readonly bool IsLittleEndian;
 
+#if !MOBILE // Should be PLATFROM_LINUX instead
 		static BitConverter () {
```
Comment (vargaz) [inline mcs/class/corlib/System/BitConverter.cs]: Why was this added ? The cctor is needed to compute IsLittleEndian.

Comment (marek-safar) [reply mcs/class/corlib/System/BitConverter.cs]: Yeah, I keep forgetting about the inverted logic
